### PR TITLE
Add debug request parameters

### PR DESCRIPTION
**Pull request by aw-leigh**
### Purpose/goal of PR (Issue #)
https://wovnio.atlassian.net/browse/ST-589

### Comments
Adds support for `wovnDisable` URL parameter which if enabled will cause the library to not swap the request.

Also adds a Debug Mode setting that enables two URL parameters:
`wovnCacheDisable`: inserts a timestamp into the API call url to circumvent Fastly caching
`wovnDebugMode`: adds {“debug_mode”: “true”} to API call request payload, and add a timestamp to the API call to circumvent Fastly cache.

Usage:
All parameters can enabled by adding to the URL either without a value (e.g. `?lang=ja&wovnCacheDisable`) or with any value other than the string `'false'` (e.g. `?lang=ja&wovnDebugMode=true`).

### CODE REVIEW

File: src/wovn_interceptor.php [hunk 1]
```diff
@@ -20,10 +21,15 @@
...
 use Wovnio\Wovnphp\API;
 use Wovnio\Wovnphp\Diagnostics;
+use Wovnio\Wovnphp\RequestOptions;
 use \Wovnio\Wovnphp\CookieLang;
 
 // GET STORE AND HEADERS
 list($store, $headers) = Utils::getStoreAndHeaders($_SERVER, $_COOKIE);
+// RequestOptions here?
+
+$queryString = parse_url($headers->getEnv()['REQUEST_URI'])['query'];
```
Comment (bstewart00) [inline src/wovn_interceptor.php:31]: yes, request options here but you can use $_SERVER directly instead of headers->getEnv

Comment (bstewart00) [reply src/wovn_interceptor.php:31]: well, I guess we could make $headers->query public as well, regrettably

**Comment by aw-leigh** [comment]
Since API::translate is just a static function call here (as opposed to a method in an API object like in Wovnjava), I added a `$cache_disable_mode` parameter. Is this alright or is there a better way I'm overlooking? If it's alright then I'll have to edit the calls in APITest, but I wanted to double check before doing so. Or I guess I could make `$cache_disable_mode` false by default, but I'm told that default parameters are to be avoided if possible.

**Comment by bstewart00** [comment]
yes, using a parameter instead is the usual method

File: src/wovnio/wovnphp/RequestOptions.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+
+    public function __construct($queryString, $debugModeEnable)
+    {
+        $this->disableMode = false;
+        $this->cacheDisableMode = false;
+        $this->debugMode = false;
+
+        if ($queryString !== null) {
+            $this->disableMode = strpos($queryString, 'wovnDisable') !== false;
```
Comment (bstewart00) [inline src/wovnio/wovnphp/RequestOptions.php:35]: note that this is a bit naive. We would like to allow `wovnDisable=false/true` ideally. Which is why making `$headers->query` public (+ https://www.php.net/manual/en/function.parse-str.php perhaps to turn it into an array) might be a better approach.

Comment (aw-leigh) [reply src/wovnio/wovnphp/RequestOptions.php:35]: I can do that, though I'm not sure when we'd ever want to explicitly set wovnDisable to false.

Comment (bstewart00) [reply src/wovnio/wovnphp/RequestOptions.php:35]: It's handy when you're debugging to switch between on/off quickly

Comment (bstewart00) [reply src/wovnio/wovnphp/RequestOptions.php]: Didn't realize wovnjava also behaves like this though.

Comment (aw-leigh) [reply src/wovnio/wovnphp/RequestOptions.php]: Yeah that's why I did it that way 🙃

**Review by zeyuwu — APPROVED** [review]

**Review by bstewart00 — APPROVED** [review]

**Pull request merged**